src/MainWindow.vala

```diff
@@ -55,6 +58,10 @@ public class Camera.MainWindow : Hdy.ApplicationWindow {
     public MainWindow (Application application) {
         Object (application: application);
 
+        change_camera_action = new SimpleAction.stateful (ACTION_CHANGE_CAMERA, new VariantType ("s"), new Variant.string (""));
+        change_camera_action.activate.connect (on_change_camera);
+
+        add_action (change_camera_action);
```

Comment:
This can be another entry in the `ACTION_ENTRIES` array, you can use `change_action_state()` to change the state.